sklearn/datasets/base.py

```diff
@@ -57,8 +57,16 @@ def __getattr__(self, key):
         except KeyError:
             raise AttributeError(key)
 
-    def __getstate__(self):
-        return self.__dict__
```

Comment:
I don't think `__getstate__` was ever necessary. According to [this](https://groups.google.com/forum/#!topic/python-ideas/QkvOwa1-pHQ) non overriding `__getstate__` is the same as returning `self.__dict__.copy()`.